naomi-setup.sh

```diff
@@ -7,54 +7,239 @@
+    # requires ssl for all connections.
+    NAME=pip
+    VERSION=18.1
+    VERNAME=$NAME-$VERSION
+    CHKSUM=c0a292bd977ef590379a3f05d7b7f65135487b67470f6281289a94e015650ea1
+    TARFILE=$VERNAME.tar.gz
+    URL=https://files.pythonhosted.org/packages/45/ae/8a0ad77defb7cc903f09e551d88b443304a9bd6e6f124e75c0fbbf6de8f7/$TARFILE
+    if [ ! -f $TARFILE ]; then
+        wget $URL
+    fi
+    echo "$CHKSUM $TARFILE" | sha256sum -c -
+    if [ $? -eq 0 ]; then
+        echo "$TARFILE checksum verified"
+    else
+        echo "$TARFILE checksum failed" >&2
+        exit 1
+    fi
+    tar xvzf $TARFILE
+    cd $VERNAME # ~/.naomi/local/pip-18.1
+    ~/.naomi/local/bin/python setup.py install  # specify the path to the python you installed above
 
-# The old version of pip expects to access pypi.org using http. PyPi now
-# requires ssl for all connections.
-wget https://files.pythonhosted.org/packages/45/ae/8a0ad77defb7cc903f09e551d88b443304a9bd6e6f124e75c0fbbf6de8f7/pip-18.1.tar.gz
-tar xvzf pip-18.1.tar.gz
-cd pip-18.1 # ~/.naomi/local/pip-18.1
-~/.naomi/local/bin/python setup.py install  # specify the path to the python you installed above
+    # install naomi & dependancies
+    echo "Returning to $NAOMI_DIR"
+    cd $NAOMI_DIR
+    ~/.naomi/local/bin/pip install -r python_requirements.txt
 
-# install naomi & dependancies
-echo "Returning to $NAOMI_DIR"
-cd $NAOMI_DIR
-~/.naomi/local/bin/pip install -r python_requirements.txt
+    # start the naomi setup process
+    echo "#!/bin/bash" > Naomi
+    echo "~/.naomi/local/bin/python $NAOMI_DIR/Naomi.py \$@" >> Naomi
+fi
+if [ $OPTION = "3" ]; then
+    if [ $APT -eq 1 ]; then
+        echo 'Making sure you have the latest python3 and pip3 installed on your system'
+        sudo apt-get install python3 python3-pip
+    else
+        echo "Please ensure that python3 and pip3 are installed on your system"
+    fi
+    pip3 install -r python_requirements.txt
+    # start the naomi setup process
+    echo "#!/bin/bash" > Naomi
+    echo "python3 $NAOMI_DIR/Naomi.py \$@" >> Naomi
+fi
 ./compile_translations.sh
-
-# start the naomi setup process
-echo "#!/bin/bash" > Naomi
-echo "~/.naomi/local/bin/python $NAOMI_DIR/Naomi.py \$@" >> Naomi
 chmod a+x Naomi
+if [ $OPTION = "1" ]; then
+    echo "You will need to activate the Naomi virtual environment when installing"
+    echo "or testing python modules for Naomi using the following command:"
+    echo "  $ workon Naomi"
+    echo "You should add the following lines to your ~/.bashrc script:"
+    echo "  export VIRTUALENVWRAPPER_VIRTUALENV=~/.local/bin/virtualenv"
+    echo "  source ~/.local/bin/virtualenvwrapper.sh"
+    echo
+fi
+if [ $OPTION = "2" ]; then
+    echo "You will need to use Naomi's special python and pip commands when"
+    echo "installing modules or testing with Naomi:"
+    echo "  ~/.naomi/local/bin/python"
+    echo "  ~/.naomi/local/bin/pip"
+    echo
+fi
 echo "In the future, run $NAOMI_DIR/Naomi to start Naomi"
```

Comment:
It might be a great idea if we can be coloring text whenever possible. 

RED='\033[0;31m'
NC='\033[0m' # No Color
printf "${RED}In the future, run $NAOMI_DIR/Naomi to start Naomi ${NC}"

This shouldn't break in terminals that don't colors but for those do, it'll work great...

Red might be bad choice. So here is a color list.
```
Black        0;30     Dark Gray     1;30
Red          0;31     Light Red     1;31
Green        0;32     Light Green   1;32
Brown/Orange 0;33     Yellow        1;33
Blue         0;34     Light Blue    1;34
Purple       0;35     Light Purple  1;35
Cyan         0;36     Light Cyan    1;36
Light Gray   0;37     White         1;37
```

Reply:
I've used a little color and put all of the above definitions in which should make it easier for others to add colors.